src/nnbench/types.py

```diff
@@ -6,10 +6,17 @@
 import copy
 import inspect
 import os
+import shutil
+import weakref
 from abc import ABCMeta, abstractmethod
 from dataclasses import dataclass, field
+from pathlib import Path
+from tempfile import mkdtemp
 from typing import Any, Callable, Generic, Iterable, Iterator, Literal, TypeVar
 
+from fsspec import core, filesystem, utils
```

Comment:
Please import the specific APIs from these submodules, not the whole modules themselves.